icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -271,14 +273,13 @@ void U_CALLCONV initUnitExtras(UErrorCode& status) {
 
     // Build the CharsTrie
     // TODO: Use SLOW or FAST here?
-    UnicodeString result;
-    b.buildUnicodeString(USTRINGTRIE_BUILD_FAST, result, status);
+    StringPiece result = b.buildStringPiece(USTRINGTRIE_BUILD_FAST, status);
     if (U_FAILURE(status)) { return; }
 
     // Copy the result into the global constant pointer
-    size_t numBytes = result.length() * sizeof(char16_t);
-    kSerializedUnitExtrasStemTrie = static_cast<char16_t*>(uprv_malloc(numBytes));
-    uprv_memcpy(kSerializedUnitExtrasStemTrie, result.getBuffer(), numBytes);
+    size_t numBytes = result.length();
+    kSerializedUnitExtrasStemTrie = static_cast<char *>(uprv_malloc(numBytes));
```

Comment:
Technically unrelated to this specific change/PR, but `uprv_malloc`/`malloc` can fail for some platforms (due to out-of-memory conditions) and return `NULL`/`nullptr`.

If you don't mind, I think it would be good to check for `nullptr` here, and set the error code to `U_MEMORY_ALLOCATION_ERROR`. :)

Perhaps something like this:
```c++
if (kSerializedUnitExtrasStemTrie == nullptr) {
    status = U_MEMORY_ALLOCATION_ERROR;
    return;
}
uprv_memcpy(kSerializedUnitExtrasStemTrie, result.data(), numBytes);
```

Reply:
Yeah, good catch

Reply:
Done!

Reply:
Thanks!